File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
import java.util.*;
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        StringBuilder sb = new StringBuilder();
        for (int i=0; i<n; ++i){
            for (int j=0; j<n; ++j){
                String binary1 = String.format("%0"+n+"d", Long.parseLong(Long.toBinaryString(arr1[i])));
                String binary2 = String.format("%0"+n+"d", Long.parseLong(Long.toBinaryString(arr2[i])));
                if (binary1.charAt(j) == 49 || binary2.charAt(j) == 49){
                    sb.append("#");
                } else sb.append(" ");
            }
            answer[i] = sb.toString();
            sb.setLength(0);
            System.out.println(Arrays.toString(answer));
        }


        return answer;
    }
}

File: JavaScript/프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
function solution(n, arr1, arr2) {
    let answer = [];

    for (let i = 0; i < n; i++) {
        let str1 = arr1[i].toString(2).padStart(n, '0');
        let str2 = arr2[i].toString(2).padStart(n, '0');
        let row = '';

        for (let j = 0; j < n; j++) {
            row += str1[j] == '1' || str2[j] == '1' ? '#' : ' ';    
        }
        answer.push(row);
    }
    return answer;
}